article/article.Rmd

```diff
@@ -0,0 +1,224 @@
+
+-   unspecified date( component)s, e.g. `2012-XX-01` for the first of some unknown month in 2012 or `2012-01` for some unknown day in January 2012
+-   approximate date( component)s, e.g. `2012-01-12~` for approximately the 12th of January 2012
+-   uncertain date( component)s, e.g. `2012-01-12?` where this data point is based on an unreliable source
+-   sets of dates, e.g. `{2012-01-01,2012-01-12}` where the date can be both 1 January 2012 and 12 January 2012
+-   ranges of dates, e.g. `2012-01-01..2012-01-12` for all dates between the 1 January 2012 and 12 January 2012 inclusive
+
+\pkg{messydates} contains a set of tools for constructing and coercing into and from the `mdate` class. This date class allows regular dates to be annotated to express unspecified date components, approximate or uncertain date components, date ranges, and sets of dates.
+
+Importantly, the package also includes a function for unpacking or expanding sets or ranges of dates into all dates consistent with how the date or set of dates is specified or annotated. Methods are also offered that can be used to make explicit how researchers convert date imprecision into precise dates for analysis, such as getting the `min()`, `max()`, or even a `random()` date from among the dates consistent with a set or range of dates. This greatly facilitates research transparency as well as robustness checks.
+
+## Motivation
+
+As researchers, we often recognize this messiness but are forced to force non-existent precision on data so we can proceed with analysis. For example, if we only know something happened in a given month or year, we might just opt for the start of that month (e.g. `2021-07-01`) or year (`2021-01-01`), assuming that to err on the earlier (or later) side is a justifiable bias. However, this can create issues for inference in which sequence or timing is important. The goal of \pkg{messydates} is to help with this problem by retaining and working with various kinds of date imprecision.
+
+## Relationship to other packages
+
+\pkg{messydates} offers a new date class, but one that comes with methods for converting from and into common date classes such as `Date`, `POSIXct`, and `POSIXlt.` It is thus fully compatible with packages such as \pkg{lubridate} [@grolemundDatesTimesMade2011] and \pkg{anytime} [@eddelbuettelAnytimeEasierDate2019].
+
+# \proglang{R} code {#r-code short-title="R code"}
+
+## A new class
+
+\pkg{messydates} contains a set of tools for constructing and coercing into and from the `mdate` class. This date class implements ISO 8601-2:2019(E) and allows regular dates to be annotated to express unspecified date components, approximate or uncertain date components, date ranges, and sets of dates. The function `as_messydate()` handles the coercion to `mdate` class.
```

Comment:
Should we describe the motivation for and history of ISO 8601-2 somewhere?

Reply:
I added a bit more about ISO motivations, but I am not sure we need to justify further its' history.